service/lib/y2storage/proposal/agama_vg_planner.rb

```diff
@@ -28,26 +28,57 @@ module Proposal
     class AgamaVgPlanner < AgamaDevicePlanner
       # @param config [Agama::Storage::Configs::VolumeGroup]
       # @return [Array<Planned::Device>]
-      def planned_devices(config)
-        [planned_vg(config)]
+      def planned_devices(vg_config, config)
+        [planned_vg(vg_config, config)]
       end
 
     private
 
-      # @param config [Agama::Storage::Configs::VolumeGroup]
+      # @param vg_config [Agama::Storage::Configs::VolumeGroup]
+      # @param config [Agama::Storage::Config]
       # @return [Planned::LvmVg]
-      def planned_vg(config)
+      def planned_vg(vg_config, config)
         # TODO: A volume group name is expected. Otherwise, the planned physical volumes cannot
         #   be associated to the planned volume group. Should the volume group name be
         #   automatically generated if missing?
         #
         #   @see AgamaDevicePlanner#configure_pv
-        Y2Storage::Planned::LvmVg.new(volume_group_name: config.name).tap do |planned|
-          planned.extent_size = config.extent_size
-          planned.lvs = planned_lvs(config)
+        Y2Storage::Planned::LvmVg.new(volume_group_name: vg_config.name).tap do |planned|
+          planned.extent_size = vg_config.extent_size
+          planned.lvs = planned_lvs(vg_config)
+          planned.size_strategy = :use_needed
+          planned.pvs_candidate_devices = devices_for_pvs(vg_config, config)
+          configure_pvs_encryption(planned, vg_config)
         end
       end
 
+      # Names of the devices that must be used to calculate automatic physical volumes
+      # for the given volume group
+      #
+      # @param vg_config [Agama::Storage::Configs::VolumeGroup]
+      # @param config [Agama::Storage::Config]
+      # @return [Array<String>]
+      def devices_for_pvs(vg_config, config)
+        drives = vg_config.physical_volumes_devices.map do |dev_alias|
+          config.drives.find { |d| d.alias?(dev_alias) }
+        end.compact
+
+        drives.map { |d| d.found_device.name }
+      end
+
+      # Configures the encryption-related fields of the given planned volume group
+      #
+      # @param planned [Planned::LvmVg]
+      # @param config [Agama::Storage::Configs::VolumeGroup]
+      def configure_pvs_encryption(planned, config)
+        enc = config.physical_volumes_encryption
+        return unless enc
+
+        planned.pvs_encryption_method = enc.method
+        planned.pvs_encryption_password = enc.password
+        planned.pvs_encryption_pbkdf = enc.pbkd_function
```

Comment:
Are the rest of encryption attributes ignored (`#label`, `#cipher` and `#key_size`)?

Reply:
I'm not sure whether label makes sense. We are going to generate several PVs and labels should be unique.

For the other two, I overlooked because `PlannedVg` didn't support them. So it would need changes both here and at yast2-storage-ng. What about making a separate card/PBI out of it?

Reply:
Added as another point to https://trello.com/c/XAaoYdCL/578-storage-profile-improve-support-for-generating-lvm-physical-volumes